### PR TITLE
needle 0.18.1

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/needle.rb [hunk 1]
```diff
@@ -14,10 +14,21 @@ class Needle < Formula
...
+  patch do
+    url "https://github.com/uber/needle/commit/30a4f935bc5bf6ffe83e030b29c35409c72158d4.patch?full_index=1"
+    sha256 "b0692c29c37b0cd2d686ed1628894feedc675ebf3951ab3cf0ac8a8360f1b745"
+  end
+
   def install
     system "make", "install", "BINARY_FOLDER_PREFIX=#{prefix}"
     bin.install "./Generator/bin/needle"
     libexec.install "./Generator/bin/lib_InternalSwiftSyntaxParser.dylib"
+    deuniversalize_machos libexec/"lib_InternalSwiftSyntaxParser.dylib"
```
Comment (SMillerDev) [inline Formula/needle.rb:31]: Can we not patch `make` to build the correct one here?

Comment (cho-m) [reply Formula/needle.rb:31]: Makefile copies library from Xcode.app (https://github.com/uber/needle/blob/v0.17.2/Makefile#L49) and then just modifies RPATH.

Comment (SMillerDev) [reply Formula/needle.rb:31]: I don't think that allows us to ship it since XCode is proprietary

Comment (SMillerDev) [reply Formula/needle.rb:31]: We could use the one from the swift formula though. That is guaranteed open source

Comment (cho-m) [reply Formula/needle.rb]: I'll give that a try.

Comment (cho-m) [reply Formula/needle.rb]: Looks like `swift` might be taking over build from Xcode and causing some issues.

**Comment by Bo98** [comment]
There is a reason it ships with Xcode rather than the OS. `lib_InternalSwiftSyntaxParser` _must_ match the Swift version used to compile. We ship Swift 5.6 so using that dylib but still compiling using whatever older Swift version ships with older Xcodes could cause problems:

> You can either copy `lib_InternalSwiftSyntaxParser.dylib/.so` directly from the toolchain or even build it yourself from the [Swift repository](https://github.com/apple/swift), as long as you are matching the same tags or branches in both the SwiftSyntax and Swift repositories.

https://github.com/apple/swift-syntax/blob/e22754fbe767bfb290199573309689acb24297ef/Documentation/README.md#embedding-swiftsyntax-in-an-application

**Comment by cho-m** [comment]
I think the Makefile directly runs `swift` commands, so it might not be using Xcode in current state. Locally, I could build via Rosetta but not natively on M1.

Is there any extra licensing terms to allow us to just use pre-built from Xcode toolchain? I think that would reduce breakages as upstream will follow Xcode releases.

**Comment by Bo98** [comment]
> Is there any extra licensing terms to allow us to just use pre-built from Xcode toolchain?

We've done stuff like that before, like JavaNativeFoundation in openjdk@11. Sean raises a fair point but I don't really have the answer to it, because no one has really properly raised it or looked into it until now.

**Comment by SMillerDev** [comment]
Filed: https://github.com/uber/needle/issues/412

**Pull request closed without merging**